src/zenml/zen_stores/schemas/tag_schemas.py

```diff
@@ -0,0 +1,153 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""SQLModel implementation of tag tables."""
+
+
+from datetime import datetime
+from typing import TYPE_CHECKING, List
+from uuid import UUID
+
+from sqlalchemy import VARCHAR, Column
+from sqlmodel import Field, Relationship
+
+from zenml.enums import ColorVariants, TaggableResourceTypes
+from zenml.models import (
+    TagRequestModel,
+    TagResourceRequestModel,
+    TagResourceResponseModel,
+    TagResponseModel,
+    TagUpdateModel,
+)
+from zenml.zen_stores.schemas.base_schemas import BaseSchema, NamedSchema
+from zenml.zen_stores.schemas.schema_utils import build_foreign_key_field
+
+if TYPE_CHECKING:
+    from zenml.zen_stores.schemas.model_schemas import ModelSchema
+
+
+class TagSchema(NamedSchema, table=True):
+    """SQL Model for tag."""
+
+    __tablename__ = "tag"
+
+    color: str = Field(sa_column=Column(VARCHAR(255), nullable=False))
+    links: List["TagResourceSchema"] = Relationship(
+        back_populates="tag",
+        sa_relationship_kwargs={"cascade": "delete"},
+    )
+
+    @classmethod
+    def from_request(cls, request: TagRequestModel) -> "TagSchema":
+        """Convert an `TagRequestModel` to an `TagSchema`.
+
+        Args:
+            request: The request model to convert.
+
+        Returns:
+            The converted schema.
+        """
+        return cls(
+            name=request.name,
+            color=request.color.value,
+        )
+
+    def to_model(self) -> TagResponseModel:
+        """Convert an `TagSchema` to an `TagResponseModel`.
+
+        Returns:
+            The created `TagResponseModel`.
+        """
+        return TagResponseModel(
+            id=self.id,
+            name=self.name,
+            color=ColorVariants(self.color),
+            created=self.created,
+            updated=self.updated,
+            tagged_count=len(self.links),
```

Comment:
This is just a look-up of the pretty narrow table with links, I assume not a big impact vs complications it will bring to integrate that in all standard functions of sql_store, like listing for instance.